src/borg/archiver.py

```diff
@@ -1167,17 +1177,27 @@ def compare_items(output, path, item1, item2, hardlink_masters, deleted=False):
                 changes.append(compare_owner(item1, item2))
                 changes.append(compare_mode(item1, item2))
 
+            # changes is a list of paths and changesets:  [ (path1, [{changeset1}, ...]), (path2, [{changeset1}, ...]), ... ]
             changes = [x for x in changes if x]
             if changes:
-                output_line = (remove_surrogates(path), ' '.join(changes))
+                output_line = (remove_surrogates(path), changes)
 
+                # if sorting, save changes for later, otherwise go ahead and output
+                # the results as they are generated.
                 if args.sort:
                     output.append(output_line)
+                elif args.json_lines:
+                    print_json_output(output_line)
                 else:
-                    print_output(output_line)
+                    print_text_output(output_line)
 
-        def print_output(line):
-            print("{:<19} {}".format(line[1], line[0]))
+        def print_text_output(line):
+            path, diff = line
+            print("{:<19} {}".format(' '.join([txt for j, txt in diff]), path))
```

Comment:
output_line[1] used to be changes joined together by blanks, see original line 1172.
your removed the blank-joining there, but you do something different here in line 1196.

please check, guess we should not change the output format in a patch release.

Reply:
ah, ignore that. is because the element changed from text to (json, text).